### PR TITLE
Default read and reset permission for service account owners and UI enhancements

### PR DESCRIPTION
**Pull request by sarathmohan4**

### CODE REVIEW

File: tvaultapi/src/main/java/com/tmobile/cso/vault/api/service/ServiceAccountsService.java [hunk 1]
```diff
@@ -1136,28 +1136,65 @@ private void sleep() {
...
-										put(LogMessage.MESSAGE, "Metadata update Success.").
-										put(LogMessage.STATUS, metadataResponse.getHttpstatus().toString()).
-										put(LogMessage.APIURL, ThreadLocalContext.getCurrentMap().get(LogMessage.APIURL).toString()).
-										build()));
-							}
-							else {
-								log.debug(JSONUtil.getJSON(ImmutableMap.<String, String>builder().
+
+							// Check metadata to get the owner information
+							Response getMetaResponse = getMetadata(token, userDetails, TVaultConstants.SVC_ACC_ROLES_PATH + svcAccName);
```
Comment (sivanag1974) [inline tvaultapi/src/main/java/com/tmobile/cso/vault/api/service/ServiceAccountsService.java]: variable naming problem

File: tvaultapi/src/main/java/com/tmobile/cso/vault/api/service/ServiceAccountsService.java [hunk 2]
```diff
@@ -1136,28 +1136,65 @@ private void sleep() {
...
+							try {
+								JsonNode metaNode = new ObjectMapper().readTree(getMetaResponse.getResponse()).get("data").get("initialPasswordReset");
```
Comment (sivanag1974) [inline tvaultapi/src/main/java/com/tmobile/cso/vault/api/service/ServiceAccountsService.java]: What if getMetaResponse is null?

File: tvaultapi/src/main/java/com/tmobile/cso/vault/api/service/ServiceAccountsService.java [hunk 3]
```diff
@@ -1136,28 +1136,65 @@ private void sleep() {
...
+								if (metaNode != null) {
+									boolean initialResetStatus = false;
+									if (metaNode != null) {
```
Comment (sivanag1974) [inline tvaultapi/src/main/java/com/tmobile/cso/vault/api/service/ServiceAccountsService.java]: what is the use of the check "if (metaNode != null) {" two times?

File: tvaultapi/src/main/java/com/tmobile/cso/vault/api/service/ServiceAccountsService.java [hunk 4]
```diff
@@ -1136,28 +1136,65 @@ private void sleep() {
...
+										 initialResetStatus = Boolean.parseBoolean(metaNode.asText());
+										 if (!initialResetStatus) {
+
+											 // update metadata for initial password reset
+											 String path = new StringBuffer(TVaultConstants.SVC_ACC_ROLES_PATH).append(svcAccName).toString();
+											 Map<String,String> params = new HashMap<String,String>();
```
Comment (sivanag1974) [inline tvaultapi/src/main/java/com/tmobile/cso/vault/api/service/ServiceAccountsService.java]: HashMap - is not synchronized

File: tvaultapi/src/main/java/com/tmobile/cso/vault/api/service/ServiceAccountsService.java (hunk 1 above)
Comment (sivanag1974) [inline tvaultapi/src/main/java/com/tmobile/cso/vault/api/service/ServiceAccountsService.java]: Variable naming

File: tvaultapi/src/main/java/com/tmobile/cso/vault/api/service/ServiceAccountsService.java (hunk 2 above)
Comment (sivanag1974) [inline tvaultapi/src/main/java/com/tmobile/cso/vault/api/service/ServiceAccountsService.java]: what is getMetaResponse is null

File: tvaultapi/src/main/java/com/tmobile/cso/vault/api/service/ServiceAccountsService.java (hunk 1 above)
Comment (sarathmohan4) [reply tvaultapi/src/main/java/com/tmobile/cso/vault/api/service/ServiceAccountsService.java]: corrected

File: tvaultapi/src/main/java/com/tmobile/cso/vault/api/service/ServiceAccountsService.java (hunk 3 above)
Comment (sarathmohan4) [reply tvaultapi/src/main/java/com/tmobile/cso/vault/api/service/ServiceAccountsService.java]: removed

**Pull request merged**